feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/service/storage/impl/MetisStorageServiceImplTest.kt

```diff
@@ -0,0 +1,165 @@
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@Category(UnitTest::class)
+@RunWith(RobolectricTestRunner::class)
+class MetisStorageServiceImplTest {
+
+    private val databaseProviderMock = MetisDatabaseProviderMock(InstrumentationRegistry.getInstrumentation().context)
+    private val sut = MetisStorageServiceImpl(databaseProviderMock)
+
+    private val host = "host"
+
+    private val author = User(id = 20, name = "AuthorName")
+    private val parentClientPostId = "parent-client-id-0"
+    private val answerClientPostId = "answer-client-id-0"
+
+    private val course: MetisContext.Course = MetisContext.Course(courseId = 1)
+    private val conversation = OneToOneChat(id = 2)
+    private val metisContext = MetisContext.Conversation(course.courseId, conversation.id)
+
+    private val localAnswerPojo = AnswerPostPojo(
+        parentPostId = parentClientPostId,
+        postId = answerClientPostId,
+        resolvesPost = false,
+        basePostingCache = AnswerPostPojo.BasePostingCache(
+            serverPostId = 0,
+            authorId = author.id,
+            creationDate = Clock.System.now(),
+            updatedDate = null,
+            content = "Answer post content 0",
+            authorRole = UserRole.USER,
+            authorName = author.name!!
+        ),
+        reactions = emptyList(),
+        serverPostIdCache = AnswerPostPojo.ServerPostIdCache(
+            serverPostId = null     // Only local answer post, no server id
+        )
+    )
+
+    private val basePostPojo = PostPojo(
+        clientPostId = parentClientPostId,
+        serverPostId = 0,
+        content = "Base post content",
+        resolved = false,
+        updatedDate = null,
+        creationDate = Clock.System.now(),
+        authorId = author.id,
+        title = null,
+        authorName = author.name!!,
+        authorRole = UserRole.USER,
+        courseWideContext = null,
+        tags = emptyList(),
+        answers = emptyList(),
+        reactions = emptyList()
+    )
+
+    private val basePost = StandalonePost(basePostPojo, conversation)
+    private val localAnswer = AnswerPost(localAnswerPojo, basePost)
+
+    @Test
+    fun testInsertClientSidePost() = runTest {
+        // GIVEN: A base post
+        sut.insertOrUpdatePosts(
+            host = host,
+            metisContext = metisContext,
+            posts = listOf(basePost),
+        )
+
+        // WHEN: Inserting a client side answer post
+        sut.insertClientSidePost(
+            host = host,
+            metisContext = metisContext,
+            post = localAnswer,
+            clientSidePostId = answerClientPostId
+        )
+
+        // THEN: Both the base post and the answer post are stored
+        val posts = sut.getStoredPosts(
+            serverId = host,
+            metisContext = metisContext
+        ).loadAsList()
+
+        assertEquals(1, posts.size)
+        assertEquals(1, posts.first().answers.size)
```

Comment:
Please add assertions for the data within the retrieved posts. This just checks that a post with an answer is present, but not that they also hold identical data to what was previously inserted.